onnxruntime/python/tools/kernel_explorer/device_array.h

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "hip/hip_runtime.h"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/numpy.h>
+
+namespace py = pybind11;
+
+#define HIP_ASSERT(x) (assert((x)==hipSuccess))
```

Comment:
this is a Hip array? could we make it also works with cuda?

Reply:
Yeah, I thought about that a bit, it will add some complexity to the code either with hipify or with some macros as a more lightweight solution (different header files need to be included, etc). As the tool is still experimental and going through frequent changes, it might make sense to not bear that extra complexity for now. But at some point down the road, I think we do need to make this more general for both CUDA and HIP.